crates/codegen/language/definition/src/model/terminals/scanner.rs

```diff
@@ -40,3 +40,14 @@ pub enum Scanner {
         reference: Identifier,
     },
 }
+
+impl Scanner {
+    pub fn is_unique(&self) -> bool {
+        match self {
+            Self::Sequence { scanners } => scanners.iter().all(|scanner| scanner.is_unique()),
+            Self::TrailingContext { scanner, .. } => scanner.is_unique(),
+            Self::Atom { .. } => true,
+            _ => false,
```

Comment:
nit suggestion: listing all variations explicitly, to force us to review this list whenever the variations change.